lib/view_component/base.rb

```diff
@@ -64,20 +63,22 @@ def render_in(view_context, &block)
       # For template variants (+phone, +desktop, etc.)
       @variant = @lookup_context.variants.first
 
+      # Assign captured content passed to component as a block to @content
+      @content = view_context.capture(self, &block) if block_given?
+
       # For caching, such as #cache_if
       @current_template = nil unless defined?(@current_template)
       old_current_template = @current_template
       @current_template = self
 
-      # Assign captured content passed to component as a block to @content
-      @content = view_context.capture(self, &block) if block_given?
+      return "" unless render?
+
+      instrument("render_template") do
+        self.class.compile(raise_errors: true)
```

Comment:
I was not sure about this. So I may do the change.

I have done this that way because `action_view` capture compilation. But may be in our case it's not relevant.
(https://github.com/rails/rails/blob/528b62e3869288b5f3ffbd1491f10bad00958c08/actionview/lib/action_view/template.rb#L180-L183)